frontend/src/components/Checklists.tsx

```diff
@@ -0,0 +1,104 @@
+import React, { Fragment, useContext } from 'react';
+import {
+  Typography,
+  List,
+  ListItem,
+  Box,
+  Chip,
+  Checkbox,
+} from '@material-ui/core';
+import { Module } from '../types';
+import { AppContext } from '../context';
+import CheckCircleOutlineIcon from '@material-ui/icons/CheckCircleOutline';
+import PanoramaFishEyeIcon from '@material-ui/icons/PanoramaFishEye';
+import { useStyles, } from '../styles';
+import ReactMarkdown from 'react-markdown';
+import {
+  getSupportedTools,
+  getSelectedTools,
+} from '../utils/moduleHelpers';
+
+interface Props {
+  module: Module;
+  readOnlyMode: boolean;
+}
+interface ChecklistProps {
+  toolsSupported: string[];
+  readOnlyMode: boolean;
+}
+
+const ListoCheckbox = (props: ChecklistProps) => {
+  const classes = useStyles();
+  if (props.toolsSupported.length > 0) {
+    return (
+      <div className={classes.checklistQuestion}>
+        <CheckCircleOutlineIcon className={classes.questionIcon} />
+      </div>
+      );
+  } else if (!props.readOnlyMode) {
+    return ( <Checkbox color="primary" className={classes.checkboxIcon} />);
+  }
+  return (
+    <div className={classes.checklistQuestion}>
+      <PanoramaFishEyeIcon className={classes.questionIcon} />
+    </div>
+  );
+};
+
+const Checklists = ({
+  module,
+  readOnlyMode
+}: Props) => {
+  const { tools } = useContext(AppContext);
+  const classes = useStyles();
+
+  const selectedTools = getSelectedTools(tools);
+
+  return (
+    <List dense={true}>
+      {Object.entries(module.checkLists).map(
+        ([checklistName, checklistItems]) => {
+          return (
+            <Fragment key={checklistName}>
+              <Typography>{checklistName}</Typography>
+              {checklistItems.map(checklistItem => {
+                const toolsSupported = getSupportedTools(
+                  checklistItem,
+                  selectedTools,
+                );
+                return (
+                  <ListItem key={checklistItem.question}>
+                    <ListoCheckbox readOnlyMode={readOnlyMode} toolsSupported={toolsSupported} />
+                    <Box>
+                      <ReactMarkdown source={checklistItem.question} />
+                      <div className={classes.toolsWrapper}>
+                        {toolsSupported.length > 0 ? toolsSupported.map((tool, index) => (
```

Comment:
You don't need this length check

Reply:
Do you also need the read-only check here?

If you do,  you can smoosh these down:

```
{[...toolsSupported, ...checklistItem.tools].map(tool => {
  <Chip
    key={index}
    className={classes.toolChip}
    label={tool}
  />
}}
```

Reply:
I could figure out a way to simplify this for sure. However we do need the readonly check as this component is used to display the checklists within the module selection and the QuickChecklists feature. The latter does not have the concept of tool selection, so added this second Chip to let users know what tools would do this for them (i.e. display all the tools associated with the checklist item).